src/catalog/database_catalog.cpp

```diff
@@ -953,11 +964,154 @@ const Schema &DatabaseCatalog::GetSchema(const common::ManagedPointer<transactio
+                                         const IndexSchema &schema) {
+  if (!TryLock(txn)) return INVALID_CONSTRAINT_OID;
+  const constraint_oid_t constraint_oid_t = static_cast<::terrier::catalog::constraint_oid_t>(next_oid_++);
+  return CreateConstraintsEntry(txn, ns, table, constraint_oid_t, name, schema) ? constraint_oid_t : INVALID_CONSTRAINT_OID;
+}
+
+bool DatabaseCatalog::CreateConstraintsEntry(const common::ManagedPointer<transaction::TransactionContext> txn,
+                                       const namespace_oid_t ns_oid, const table_oid_t table_oid,
+                                       const constraint_oid_t constraint_oid, const std::string &name,
+                                       const IndexSchema &schema) {
+  // Insert metadata into pg_constraint
+  auto *const constraints_insert_redo = txn->StageWrite(db_oid_, postgres::CONSTRAINT_TABLE_OID, pg_constraints_all_cols_pri_);
+  auto *const constraints_insert_pr = constraints_insert_redo->Delta();
+
+  // Write the constraint_oid into the PR
+    auto con_oid_offset = pg_constraints_all_cols_prm_[postgres::CONOID_COL_OID];
+    auto *con_oid_ptr = constraints_insert_pr->AccessForceNotNull(con_oid_offset);
+    *(reinterpret_cast<constraint_oid_t *>(con_oid_ptr)) = constraint_oid;
+
+  // Write the constraint_name into the PR
+    const auto name_varlen = storage::StorageUtil::CreateVarlen(name);
+    auto con_name_offset = pg_constraints_all_cols_prm_[postgres::CONNAME_COL_OID];
+    auto con_name_ptr = constraints_insert_pr->AccessForceNotNull(con_name_offset);
+    *(reinterpret_cast<storage::VarlenEntry *>(con_name_ptr)) = name_varlen;
+
+  // Write the namespace_oid into the PR
+  const auto con_namespace_oid_offset = pg_constraints_all_cols_prm_[postgres::CONNAMESPACE_COL_OID];
+  auto *const con_namespace_oid_ptr = constraints_insert_pr->AccessForceNotNull(con_namespace_oid_offset);
+  *(reinterpret_cast<namespace_oid_t *>(con_namespace_oid_ptr)) = ns_oid;
+
+  // Write constraint_type
+  const auto con_type_oid_offset = pg_constraints_all_cols_prm_[postgres::CONTYPE_COL_OID];
+  auto *const con_type_oid_ptr = constraints_insert_pr->AccessForceNotNull(con_type_oid_offset);
+  if (schema.is_unique_) {
+    *(reinterpret_cast<char *>(con_type_oid_ptr)) = 'u';
+  } else {
+    *(reinterpret_cast<char *>(con_type_oid_ptr)) = 'p';
+  }
+
+  *(reinterpret_cast<bool *>(constraints_insert_pr->AccessForceNotNull(
+      pg_constraints_all_cols_prm_[postgres::CONDEFERRABLE_COL_OID]))) = false;
+  *(reinterpret_cast<bool *>(constraints_insert_pr->AccessForceNotNull(
+      pg_constraints_all_cols_prm_[postgres::CONDEFERRED_COL_OID]))) = false;
+  *(reinterpret_cast<bool *>(constraints_insert_pr->AccessForceNotNull(
+      pg_constraints_all_cols_prm_[postgres::CONVALIDATED_COL_OID]))) = true;
+
+//  const auto con_relid_oid_offset = pg_constraints_all_cols_prm_[postgres::CONRELID_COL_OID];
```

Comment:
Not sure if this code is going to be reused in the future? Might consider removing it if it is dead code?